main.rb

```diff
@@ -0,0 +1,470 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # My each method definition
+
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    arr = self
+    if arr.is_a? Array
+      0.upto(arr.length - 1) do |index|
+        yield (arr[index])
+      end
+    elsif arr.is_a? Range
+      arr.step(1) do |value|
+        yield value
+      end
+    end
+    arr
+  end
+
+  # my_each_with_index method definition
+  def my_each_with_index
+    return to_enum :my_each unless block_given?
+
+    arr = self
+    0.upto(arr.length - 1) do |ii|
+      yield arr[ii], ii
+    end
+    arr
+  end
+
+  # my_select method definition
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    arr = self
+    back = []
+    arr.my_each { |x| back << x if yield x }
+    back
+  end
+
+  def param_reg(par, arr_i, t_reg, t_patt, who)
+    # who is false for my_all method call
+    # who is true for my_none method call
+    return [t_reg, t_patt] unless par != ''
+
+    if par.class == Regexp && (!par.match(arr_i).nil? ^ who)
+      t_reg += 1
+    elsif par.class != Regexp && arr_i.is_a?(Module.const_get(par.to_s))
+      t_patt += 1
+    end
+    [t_reg, t_patt]
+  end
+
+  # my_all? method definition
+  def my_all?(param = '')
+    arr = self
+    any_h = { t_b_e: 0, t_e: 0, t_rxp: 0, t_p: 0, who: false }
+
+    0.upto(arr.length - 1) do |i|
+      if param == ''
+        if block_given?
+          any_h[:t_b_e] += 1 if yield arr[i]
+        elsif !arr[i].nil? && arr[i] != false
+          any_h[:t_e] += 1
+        end
+      end
+      any_h[:t_rxp], any_h[:t_p] = param_reg(param, arr[i], any_h[:t_rxp], any_h[:t_p], false)
+    end
+    any_element_true(arr.length, any_h[:t_b_e], any_h[:t_e], any_h[:t_rxp], any_h[:t_p])
+
+    # if we reach this step and all elements are thruty, we exit true.
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
```ruby
array.my_all?(3) == array.all?(3) #true
```

Reply:
Got it. Thanks for the hint. This is fixed now.